Modules/IO/NIFTI/src/itkNiftiImageIO.cxx

```diff
@@ -1543,7 +1548,11 @@ NiftiImageIO::WriteImageInformation()
     }
     else
     {
+      // TODO: There is no way to tell if it is a displacement vector (NIFTI_INTENT_DISPVECT)
+      // or some other vector (NIFTI_INTENT_VECTOR), therefore we always use generic vector intent.
       this->m_NiftiImage->intent_code = NIFTI_INTENT_VECTOR;
+      // TODO: consider reorienting vector field values (NIFTI_INTENT_DISPVECT or NIFTI_INTENT_VECTOR)
+      // from ITK's LPS coordinate system to NIFTI's native RAS coordinate system.
```

Comment:
Thanks @lassoan Would it be and idea to add such information to the MetaDataDictionary of the image? For example:

```
  // Assume LPS if unspecified.
  EncapsulateMetaData(
    metaDataDictionary, "CoordinateSystem", SpatialOrientationEnums::ValidCoordinateOrientations::ITK_COORDINATE_ORIENTATION_RAS);
  EncapsulateMetaData(metaDataDictionary, "Intent", "DisplacementField");

```

Reply:
Of course, the writer then needs to make of the information in MetaDataDictionary.

Reply:
Thanks for the suggestions!

MetaDataDictionary could work well for reading/writing the intent code. I'll add a commit to this PR to use these dictionary fields.

For the vector reorientation, I see two main options:
- Option A: ITK always converts to/from RAS if intent is "displacement vector" (does not reorient if intent is just generic "vector"). This would make all ITK-based tools handle "displacement vector" data sets as expected. Helper functions for converting between RAS/LPS coordinate system may be made available to users to manually deal with data sets that use generic "vector" intent for storing spatial data.
- Option B: Let the user reorient (regardless of "displacement vector" or "vector" intent), based on the MetadataDictionary field. This would be simpler for ITK but most likely users/applications would not be aware that flipping is necessary and they would just work incorrectly.

Would you have any preference?

Reply:
Based on your description, I prefer A.

Reply:
I think I would prefer option A as well, unless some users would prefer to remain in RAS space...? If option A can be implemented, of course my suggested `("CoordinateSystem", ITK_COORDINATE_ORIENTATION_RAS)` dictionary entry would not be necessary anymore. I don't know if an `("Intent", "DisplacementField")` dictonary entry would then still be useful either. No problem.

Reply:
I was thinking about the automatic conversion to RAS, too. Slicer currently uses RAS internally, so after ITK reads a displacement field from a NRRD file, Slicer convert all the vectors to RAS. In case of NIFTI it would mean two conversions - RAS->LPS in ITK then LPS->RAS in Slicer. The time needed for conversion is usually not noticeable, so we could implement this simple, always-convert solution, and then if somebody runs into a performance issue then a bypass mechanism could be implemented.

Reply:
The ITK MINC reader needs this LPS/RAS conversion as it currently blindly assumes LPS, which is wrong, see #147